src/test/java/org/commcare/mockapp/AsyncDataCollisionTests.java

```diff
@@ -0,0 +1,164 @@
+package org.commcare.mockapp;
+
+import org.commcare.backend.suite.model.test.EmptyAppElementsTests;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.SessionNavigator;
+import org.commcare.suite.model.MenuDisplayable;
+import org.commcare.suite.model.MenuLoader;
+import org.commcare.test.utilities.MockApp;
+import org.commcare.test.utilities.MockSessionNavigationResponder;
+import org.commcare.test.utilities.XFormTestUtilities;
+import org.commcare.util.LoggerInterface;
+import org.commcare.util.mocks.JavaPlatformFormSubmitTool;
+import org.commcare.util.mocks.JavaPlatformSyncTool;
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.core.model.data.IntegerData;
+import org.javarosa.core.util.externalizable.LivePrototypeFactory;
+import org.javarosa.form.api.FormEntryController;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.text.Normalizer;
+
+import static org.junit.Assert.assertEquals;
+
+/**
```

Comment:
note: This is not a proper unit test, but will be run with the unit tests right now. It should get fixed to run only during some sort of integration test cycle.